projects/web-wallet/wallet-connect/0001-design.md

```diff
@@ -0,0 +1,774 @@
+            ],
+            "events": [],
+            "methods": [
+                "hathor_signMessage"
+            ]
+        }
+    }
+}
+```
+
+Once the approve_session message is received and processed by the dApp, the secure communication channel between the dApp and the wallet is established, and they can exchange encrypted messages through the relay server.
+
+Here is a flowchart illustrating this linking process:
+
+```mermaid
+graph TB
+  dAppGenerateQR[dApp: Generate QR Code] -- URI --> walletScanQR[Wallet: Scan QR Code]
+  walletScanQR -- Extract symKey, topic, relay info --> walletConnect[Wallet: Connect to Relay Server]
+  walletConnect -- Subscribe to topic --> walletSubscribed[Wallet: Subscribed to Topic]
+  dAppConnect[dApp: Connect to Relay Server] --> dAppSubscribed[dApp: Subscribed to Topic]
+  dAppSubscribed -- Send encrypted session_proposal --> walletSubscribed
+  walletSubscribed -- Receive session_proposal --> walletReview[Wallet: Approve or Reject]
+  walletReview -- Send encrypted approve_session --> dAppReceiveApprove[dApp: Receive approve_session]
+  dAppReceiveApprove --> dAppSecureCommunication[dApp: Establish secure communication]
+  walletReview --> walletSecureCommunication[Wallet: Establish secure communication]
+```
+
+
+**3. Message Exchange** The wallet and the dApp can now exchange encrypted messages through the relay server using the `symKey` shared through the URI (either via QRCode or by Deep-linking).
+
+**4. Session Termination** Either the wallet or the dApp can terminate the session by sending a disconnection message. The relay server then closes the WebSocket connection.
+
+#### 3. Relay Server Concepts
+
+| Field        | Description                                                                                           |
+|--------------|-------------------------------------------------------------------------------------------------------|
+| Topic        | A string identifier used by clients to publish and subscribe to messages.                             |
+| Subscription | Requested by a client and remains active until the client requests to unsubscribe.                     |
+| Publication  | Requested by a client; the Relay server is responsible for routing the message to its recipient.       |
+| Message      | A utf8 string sent when publishing, to be routed to its recipient.                                    |
+| Tag          | An integer used as metadata to identify the type of message being sent.                               |
+| TTL          | The maximum amount of time (in seconds) a message should be cached in the relay serverif undelivered.                    |
+| Prompt       | A boolean flag used to identify if a message is relevant to registered webhooks.                      |
+
+<a name="overview__is_it_secure"/>
+
+### Is it secure?
+
+The information sent between `dApps` and wallets is encrypted symmetrically with a key that only these two entities possess
```

Comment:
Is the key changed every new connection between the dApp and the wallet? Or do they always use the same key?

Reply:
It always uses the same key

I just investigated and it stores it in plain-text in the AsyncStorage (or localStorage for web)

We should extend the `core` package from their monorepo and at least encrypt it with a password as we do with the seed and private keys

I think this should also be an issue on their repository as they don't provide an API for encrypting it

Reply:
Plain text? Really? I'm questioning whether we can trust these guys' security policies. Anyway, let's add a note that it's critical to encrypt this key.

Reply:
Done in aca54e8804d6f7a907f8bd9741f5731338c6336b